CORE-2013 Added java agents to common-app plugin

woggioni opened the pull request.

Commit:
CORE-2013 Added java agents to common-app plugin

Commit:
code simplification

chrisr3 reviewed (commented):
I think we can simplify this approach.

Commit:
moved from `javaAgents.xml` to `javaAgents.properties`

added also separate example project, which is not working yet as the plugin is missing a way to set OSGi framework properties

Commit:
adapted existing client code

chrisr3 reviewed (commented):
_"Do Not Fear The Platform!"_ :owl:

Commit:
created `corda.quasar-app` Gradle plugin

Commit:
fixed framework shutdown in case of exception

mnesbit reviewed (changes requested):
We need to schedule a meeting to discuss the much smaller and simpler scope required by CORE-2013
